sensor/tests/resource/helper.go

```diff
@@ -194,74 +233,93 @@ func (c *TestContext) GetFakeCentral() *centralDebug.FakeService {
 	return c.fakeCentral
 }
 
-// RunWithResources runs the test case applying files in `files` slice in order.
-func (c *TestContext) RunWithResources(files []YamlTestFile, testCase TestCallback) {
+// run handles the run of a testRun case
+func (c *TestContext) run(t *testRun) {
+	if t.resources == nil {
+		c.runBare(t.testCase)
+	} else {
+		if t.permutation {
+			c.runWithResourcesPermutation(t)
+		} else {
+			if err := c.runWithResources(t.resources, t.testCase, t.retryCallback); err != nil {
+				c.t.Fatalf(err.Error())
+			}
+		}
+	}
+}
+
+// runWithResources runs the test case applying resources in `resources` slice in order.
+// If it is set, the RetryCallback will be called if the application of a resource fails.
+func (c *TestContext) runWithResources(resources []K8sResourceInfo, testCase TestCallback, retryFn RetryCallback) error {
 	_, removeNamespace, err := c.createTestNs(context.Background(), DefaultNamespace)
 	defer utils.IgnoreError(removeNamespace)
 	if err != nil {
-		c.t.Fatalf("failed to create namespace: %s", err)
+		return errors.Errorf("failed to create namespace: %s", err)
 	}
 	var removeFunctions []func() error
 	fileToObj := map[string]k8s.Object{}
-	for _, file := range files {
-		obj := objByKind(file.Kind)
-		removeFn, err := c.ApplyFile(context.Background(), DefaultNamespace, file, obj)
+	for i := range resources {
```

Comment:
`resources` is being modified here:

```go
removeFn, err := c.ApplyResource(context.Background(), DefaultNamespace, &resources[i], obj, retryFn)
```

The reason is because we need to set this new `Name` and that depends on whether the resource is defined as a YamlFile or and Obj. If that logic is moved out of `ApplyResource` then it would be possible to use `_, r`. WDYT?

Reply:
Today I learned that the 2nd value from `range` is copying the value. Maybe if we use a slice of references you could pass the `r` there. But I think the way you did it right now is fine.

> The reason is because we need to set this new Name and that depends on whether the resource is defined as a YamlFile or and Obj. If that logic is moved out of ApplyResource then it would be possible to use _, r. WDYT?

Up to you, really. I don't have a strong opinion here.

Reply:
I think I'll keep it as it is right now